app/main.py

```diff
@@ -1,13 +1,23 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8",
+    }
 
 
-def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+def format_single_linter_file(file_path: str, file_errors: list) -> dict:
+    return {
+        "path": file_path,
+        "errors": [format_linter_error(error) for error in file_errors],
+        "status": "passed" if not file_errors else "failed",
```

Comment:
You can avoid using `not` here
```suggestion
        "status": "failed" if file_errors else "passed",
```